ios/RNImageEditor/RNImageEditor/RNImageEditor.m

```diff
@@ -528,6 +530,67 @@ - (void)saveImageOfType:(NSString*) type folder:(NSString*) folder filename:(NSS
     }
 }
 
+- (void)saveImageWithMetadata:(NSData *)imageData fileURL:(NSURL*)fileURL
+{
+    NSString *originalFileName = [[_currentFilePath lastPathComponent] stringByDeletingPathExtension];
+    NSString *uniqueImageId = [self getUniqueImageId:originalFileName];
+
+    CGImageSourceRef source = CGImageSourceCreateWithData((CFDataRef)imageData, NULL);
+
+    NSDictionary *metadataDict = (NSDictionary *) CFBridgingRelease(CGImageSourceCopyPropertiesAtIndex(source, 0, NULL));
+    NSDictionary *exifDict = [metadataDict objectForKey:(NSString *)kCGImagePropertyExifDictionary];
+
+    NSMutableDictionary *mutableMetadataDict = [metadataDict mutableCopy];
+    NSMutableDictionary *mutableExifDict = [exifDict mutableCopy];
+
+    [mutableExifDict setValue:uniqueImageId forKey:(NSString *)kCGImagePropertyExifImageUniqueID];
+    [mutableMetadataDict setObject:mutableExifDict forKey:(NSString *)kCGImagePropertyExifDictionary];
+
+    NSMutableData *destData = [NSMutableData data];
+    CGImageDestinationRef destination =
+        CGImageDestinationCreateWithData((CFMutableDataRef)destData,CGImageSourceGetType(source), 1, NULL);
+    CGImageDestinationAddImageFromSource(destination, source, 0, (CFDictionaryRef) mutableMetadataDict);
+    CGImageDestinationFinalize(destination);
+
+    [destData writeToURL:fileURL atomically:YES];
+
+    CFRelease(destination);
+    CFRelease(source);
+
+    _currentFilePath = nil;
+}
+
+- (NSString *)getUniqueImageId:(NSString *)fileName
+{
+    NSString *reversedFileName = [self reverseString:fileName];
```

Comment:
😵‍💫

Reply:
@StasDoskalenkoHover ,
We need to extract `A57D2507383343F6B4A155B2BD06AE42` from file name: `Frame.00000.User.A57D2507383343F6B4A155B2BD06AE42.jpg` (it is for device file) and from emulated file: `img_A57D2507383343F6B4A155B2BD06AE42.jpg`.
Do you have more beautiful idea how to handle it using Objective C?